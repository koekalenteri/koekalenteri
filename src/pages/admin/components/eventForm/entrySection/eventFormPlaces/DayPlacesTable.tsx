import type { PartialEvent } from '../../types'

import { useTranslation } from 'react-i18next'
import TableCell from '@mui/material/TableCell'
import TableRow from '@mui/material/TableRow'

import { formatDate } from '../../../../../../i18n/dates'
import { getEventDays } from '../../../../../../lib/event'
import { NumberInput } from '../../../../../components/NumberInput'

import BasePlacesTable from './BasePlacesTable'

interface DayPlacesTableProps {
  event: PartialEvent
  disabled: boolean
  handleDayPlacesChange: (date: Date, value?: number) => void
  handlePlacesChange: (value?: number) => void
  totalEnabled: boolean
}

export default function DayPlacesTable({
  event,
  disabled,
  handleDayPlacesChange,
  handlePlacesChange,
  totalEnabled,
}: DayPlacesTableProps) {
  const { t } = useTranslation()
  const eventDays = getEventDays(event)

  const headers = [t('date'), 'Paikat']

  return (
    <BasePlacesTable headers={headers}>
      {eventDays.map((day) => {
        const dateStr = formatDate(day, 'yyyy-MM-dd')
        const places = event.placesPerDay?.[dateStr] || 0

        return (
          <TableRow key={dateStr}>
            <TableCell component="th" scope="row">
              {t('dateFormat.wdshort', { date: day })}
            </TableCell>
            <TableCell align="center">
              <NumberInput
                disabled={disabled || totalEnabled}
                value={places}
                onChange={(value) => handleDayPlacesChange(day, value)}
              />
            </TableCell>
          </TableRow>
        )
      })}
      <TableRow>
        <TableCell component="th" scope="row">
          Yhteens√§
        </TableCell>
        <TableCell align="center">
          <NumberInput
            id="event.places"
            disabled={disabled || !totalEnabled}
            value={event.places}
            onChange={handlePlacesChange}
          />
        </TableCell>
      </TableRow>
    </BasePlacesTable>
  )
}
