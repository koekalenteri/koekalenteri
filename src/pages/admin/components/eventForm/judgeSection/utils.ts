import type { DeepPartial, EventClass, EventType, Judge, PublicJudge } from '../../../../../types'
import type { PartialEvent } from '../../EventForm'

import { isSameDay } from 'date-fns'

export type PartialPublicJudge = Partial<PublicJudge>
export type PartialPublicJudgeValue = PartialPublicJudge | PartialPublicJudge[]

export const filterJudges = (
  judges: Judge[],
  eventJudges: PublicJudge[],
  id: number | undefined,
  eventType?: EventType
) =>
  judges
    .filter((j) => !eventType?.official || j.eventTypes.includes(eventType.eventType))
    .filter((j) => j.id === id || !eventJudges.find((ej) => ej.id === j.id))

export const hasJudge = (c: DeepPartial<EventClass>, id?: number) =>
  Array.isArray(c.judge) ? c.judge.find((j) => j.id === id) : c.judge?.id === id

export const filterClassesByJudgeId = (classes?: EventClass[], id?: number) => classes?.filter((c) => hasJudge(c, id))

export const toArray = (j?: PartialPublicJudge): PartialPublicJudge[] => (j ? [j] : [])
export const makeArray = (j?: PartialPublicJudgeValue) => (Array.isArray(j) ? [...j] : toArray(j))
export const selectJudge = (j?: PartialPublicJudgeValue, judge?: PublicJudge): PartialPublicJudge[] => {
  const a = makeArray(j)
  if (judge) {
    const index = a.findIndex((cj) => cj.id === judge.id)
    if (index === -1) {
      a.push(judge)
    } else {
      a[index] = judge
    }
  }
  return a
}

export const removeJudge = (j?: PartialPublicJudgeValue, id?: number): PartialPublicJudge[] => {
  const a = makeArray(j)
  return a.filter((cj) => cj.id !== id)
}

export const updateJudge = (
  event: PartialEvent,
  id: number | undefined,
  judge: PublicJudge | undefined,
  values?: DeepPartial<EventClass>[]
) => {
  const isSelected = (c: DeepPartial<EventClass>) =>
    values?.find((v) => event && isSameDay(v.date ?? event.startDate, c.date ?? event.startDate) && v.class === c.class)
  return event.classes?.map((c) => ({
    ...c,
    judge: isSelected(c) ? selectJudge(c.judge, judge) : removeJudge(c.judge, id),
  }))
}
