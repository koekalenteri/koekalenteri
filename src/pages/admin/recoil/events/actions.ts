import type { DogEvent } from '../../../../types'

import { useTranslation } from 'react-i18next'
import { useNavigate } from 'react-router-dom'
import { addDays, differenceInDays } from 'date-fns'
import { diff } from 'deep-object-diff'
import { useSnackbar } from 'notistack'
import { useRecoilState, useRecoilValue, useSetRecoilState } from 'recoil'

import { copyEventWithRegistrations, putEvent } from '../../../../api/event'
import { sanitizeDogEvent } from '../../../../lib/event'
import { Path } from '../../../../routeConfig'
import { eventsAtom, idTokenAtom, userSelector } from '../../../recoil'

import {
  adminEventIdAtom,
  adminNewEventAtom,
  newEventEntryEndDate,
  newEventEntryStartDate,
  newEventStartDate,
} from './atoms'
import { adminCurrentEventSelector } from './selectors'

export const useAdminEventActions = () => {
  const token = useRecoilValue(idTokenAtom)
  const user = useRecoilValue(userSelector)
  const setAdminEventId = useSetRecoilState(adminEventIdAtom)
  const [currentAdminEvent, setCurrentAdminEvent] = useRecoilState(adminCurrentEventSelector)
  const setNewEvent = useSetRecoilState(adminNewEventAtom)
  const [publicEvents, setPublicEvents] = useRecoilState(eventsAtom)
  const { enqueueSnackbar } = useSnackbar()
  const { t } = useTranslation()
  const navigate = useNavigate()

  return {
    copyCurrent,
    copyCurrentTest,
    deleteCurrent,
    save,
  }

  function updatePublicEvents(event: DogEvent, remove?: boolean): void {
    if (event.id) {
      const index = publicEvents.findIndex((e) => e.id === event.id)
      if (index >= 0) {
        const newEvents = [...publicEvents]
        if (remove) {
          newEvents.splice(index, 1)
        } else {
          newEvents.splice(index, 1, sanitizeDogEvent(event))
        }
        setPublicEvents(newEvents)
      } else if (!remove) {
        setPublicEvents([...publicEvents, sanitizeDogEvent(event)])
      }
    }
  }

  function copyCurrent() {
    if (!currentAdminEvent) {
      return
    }
    const copy = structuredClone(currentAdminEvent)
    copy.id = ''
    copy.name = 'Kopio - ' + (copy.name ?? '')
    copy.state = 'draft'
    copy.entries = 0
    copy.classes.forEach((c) => {
      c.entries = c.members = 0
      if (c.date) {
        c.date = addDays(newEventStartDate, differenceInDays(copy.startDate, c.date))
      }
      delete c.state
    })

    const days = differenceInDays(copy.endDate, copy.startDate)
    copy.startDate = newEventStartDate
    copy.endDate = addDays(newEventStartDate, days)
    copy.entryStartDate = newEventEntryStartDate
    copy.entryEndDate = newEventEntryEndDate

    delete copy.kcId
    delete copy.entryOrigEndDate

    setNewEvent(copy)
    navigate(Path.admin.newEvent)
  }

  async function copyCurrentTest() {
    if (!currentAdminEvent) {
      return
    }
    const saved = await copyEventWithRegistrations(currentAdminEvent.id, token)
    setAdminEventId(saved.id)
    setCurrentAdminEvent(saved)
    updatePublicEvents(saved)
    return saved
  }

  async function save(event: Partial<DogEvent>): Promise<DogEvent | undefined> {
    let changes: Partial<DogEvent> = event
    if (event.id && event.id === currentAdminEvent?.id) {
      const changedKeys = Object.keys(diff(currentAdminEvent, event))
      changes = { id: event.id }
      for (const key of changedKeys) {
        //@ts-expect-error typescript is just so stupid sometimes
        changes[key] = event[key]
      }
    }
    console.log(changes)
    const saved = await putEvent(changes, token)
    setAdminEventId(saved.id)
    setCurrentAdminEvent(saved)
    updatePublicEvents(saved)

    return saved
  }

  async function deleteCurrent() {
    if (!currentAdminEvent || currentAdminEvent.deletedAt) {
      return
    }

    await save({
      ...currentAdminEvent,
      deletedAt: new Date(),
      deletedBy: user?.name ?? user?.email,
    })

    updatePublicEvents(currentAdminEvent, true)

    enqueueSnackbar(t('deleteEventComplete'), { variant: 'info' })
  }
}
