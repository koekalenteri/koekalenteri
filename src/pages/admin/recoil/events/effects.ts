import type { AtomEffect } from 'recoil'
import type { DogEvent } from '../../../../types'

import { DefaultValue } from 'recoil'

import { getAdminEvents } from '../../../../api/event'
import { idTokenAtom } from '../../../recoil'

const byStartDate = (a: DogEvent, b: DogEvent) => a.startDate.valueOf() - b.startDate.valueOf()
const sortEvents = (events: DogEvent[]): DogEvent[] => [...events].sort(byStartDate)

export const adminRemoteEventsEffect: AtomEffect<DogEvent[]> = ({ getPromise, setSelf, trigger }) => {
  if (trigger === 'get') {
    setSelf(
      getPromise(idTokenAtom).then((token) =>
        token ? getAdminEvents(token).then((events) => sortEvents(events)) : new DefaultValue()
      )
      /*
      Promise.all([getPromise(idTokenAtom), getPromise(adminEventsFetchedAtom)]).then(([token, lastModified]) =>
        token ? getAdminEvents(token, lastModified).then((events) => sortEvents(events)) : new DefaultValue()
      )
      */
    )
  }
}
