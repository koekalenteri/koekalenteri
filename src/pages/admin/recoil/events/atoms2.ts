import type { DogEvent, RegistrationClass } from '../../../../types'

import { atom, atomFamily, selector } from 'recoil'

import { uniqueClasses } from '../../../../lib/utils'
import { logEffect, sessionStorageEffect } from '../../../recoil'

import { adminCurrentEventSelector, adminEventSelector } from './selectors'

export const adminEventClassAtom = atom<RegistrationClass | string>({
  key: 'adminEventClass',
  default: selector({
    key: 'adminEventClass/default',
    get: ({ get }) => {
      const event = get(adminCurrentEventSelector)
      return uniqueClasses(event)[0]
    },
  }),
  effects: [logEffect, sessionStorageEffect],
})

/**
 * Existing event editing, edits stored to session storage
 */
export const adminEditableEventByIdAtom = atomFamily<DogEvent | undefined, string>({
  key: 'adminEditableEvent/Id',
  default: adminEventSelector,
  effects: [logEffect, sessionStorageEffect],
})
