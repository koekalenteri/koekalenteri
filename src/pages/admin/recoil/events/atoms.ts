import type { GridColumnVisibilityModel } from '@mui/x-data-grid'
import type { DogEvent, RegistrationClass } from '../../../../types'

import { addDays, addYears, isSameDay, nextSaturday, sub } from 'date-fns'
import { atom, atomFamily, selector } from 'recoil'

import { zonedStartOfDay } from '../../../../i18n/dates'
import { uniqueClasses } from '../../../../lib/utils'
import { localStorageEffect, logEffect, sessionStorageEffect } from '../../../recoil'

import { adminRemoteEventsEffect } from './effects'
import { adminCurrentEventSelector, adminEventSelector } from './selectors'

export const adminEventsFetchedAtom = atom<number | undefined>({
  key: 'adminEventsFetched',
  default: addYears(new Date(), -1).valueOf(),
  effects: [logEffect, sessionStorageEffect],
})

export const adminEventsAtom = atom<DogEvent[]>({
  key: 'adminEvents',
  default: [],
  effects: [logEffect, sessionStorageEffect, adminRemoteEventsEffect],
})

const EntryStartWeeks = 6
const EntryEndWeeks = 3

export const defaultEntryStartDate = (eventStartDate: Date) => sub(eventStartDate, { weeks: EntryStartWeeks })
export const defaultEntryEndDate = (eventStartDate: Date) => sub(eventStartDate, { weeks: EntryEndWeeks })

export const newEventStartDate = zonedStartOfDay(nextSaturday(addDays(Date.now(), 90)))
export const newEventEntryStartDate = defaultEntryStartDate(newEventStartDate)
export const newEventEntryEndDate = defaultEntryEndDate(newEventStartDate)

export const isDetaultEntryStartDate = (date: Date | undefined, eventStartDate: Date) =>
  !date || isSameDay(defaultEntryStartDate(eventStartDate), date)
export const isDetaultEntryEndDate = (date: Date | undefined, eventStartDate: Date) =>
  !date || isSameDay(defaultEntryEndDate(eventStartDate), date)

export const adminNewEventAtom = atom<DogEvent>({
  key: 'newEvent',
  default: {
    state: 'draft',
    startDate: newEventStartDate,
    endDate: newEventStartDate,
    entryStartDate: newEventEntryStartDate,
    entryEndDate: newEventEntryEndDate,
    classes: [],
    judges: [{ id: 0, name: '', official: true }],
  } as unknown as DogEvent,
  effects: [logEffect, localStorageEffect],
})

export const adminShowPastEventsAtom = atom<boolean>({
  key: 'adminShowPastEvents',
  default: false,
  effects: [logEffect, localStorageEffect],
})

export const adminEventFilterTextAtom = atom<string>({
  key: 'adminEventFilterText',
  default: '',
  effects: [logEffect, localStorageEffect],
})

export const adminEventIdAtom = atom<string | undefined>({
  key: 'adminEventId',
  default: undefined,
  effects: [logEffect, localStorageEffect],
})

export const adminEventClassAtom = atom<RegistrationClass | string>({
  key: 'adminEventClass',
  default: selector({
    key: 'adminEventClass/default',
    get: ({ get }) => {
      const event = get(adminCurrentEventSelector)
      return uniqueClasses(event)[0]
    },
  }),
  effects: [logEffect, sessionStorageEffect],
})

export const adminEventColumnsAtom = atom<GridColumnVisibilityModel>({
  key: 'adminEventColumns',
  default: { id: false },
  effects: [logEffect, localStorageEffect],
})

/**
 * Existing event editing, edits stored to session storage
 */
export const adminEditableEventByIdAtom = atomFamily<DogEvent | undefined, string>({
  key: 'adminEditableEvent/Id',
  default: adminEventSelector,
  effects: [logEffect, sessionStorageEffect],
})
