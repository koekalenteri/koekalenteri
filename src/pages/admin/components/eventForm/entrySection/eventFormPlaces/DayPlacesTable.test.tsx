import { render, screen } from '@testing-library/react'

import { eventWithStaticDates } from '../../../../../../__mockData__/events'
import { flushPromises, renderWithUserEvents } from '../../../../../../test-utils/utils'

import DayPlacesTable from './DayPlacesTable'

describe('DayPlacesTable', () => {
  beforeAll(() => jest.useFakeTimers())
  afterEach(() => jest.runOnlyPendingTimers())
  afterAll(() => jest.useRealTimers())

  const eventWithPlacesPerDay = {
    ...eventWithStaticDates,
    classes: [],
    places: 10,
    placesPerDay: {
      '2021-02-10': 5,
      '2021-02-11': 5,
    },
  }

  it('should render correctly with totalEnabled=true', () => {
    const handleDayPlacesChange = jest.fn()
    const handlePlacesChange = jest.fn()

    const { container } = render(
      <DayPlacesTable
        event={eventWithPlacesPerDay}
        disabled={false}
        handleDayPlacesChange={handleDayPlacesChange}
        handlePlacesChange={handlePlacesChange}
        totalEnabled={true}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render correctly with totalEnabled=false', () => {
    const handleDayPlacesChange = jest.fn()
    const handlePlacesChange = jest.fn()

    const { container } = render(
      <DayPlacesTable
        event={eventWithPlacesPerDay}
        disabled={false}
        handleDayPlacesChange={handleDayPlacesChange}
        handlePlacesChange={handlePlacesChange}
        totalEnabled={false}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('should call handleDayPlacesChange when day places are changed', async () => {
    const handleDayPlacesChange = jest.fn()
    const handlePlacesChange = jest.fn()

    const { user } = renderWithUserEvents(
      <DayPlacesTable
        event={eventWithPlacesPerDay}
        disabled={false}
        handleDayPlacesChange={handleDayPlacesChange}
        handlePlacesChange={handlePlacesChange}
        totalEnabled={false}
      />,
      undefined,
      {
        advanceTimers: jest.advanceTimersByTime,
      }
    )
    await flushPromises()

    const inputs = screen.getAllByRole('textbox')
    expect(inputs.length).toEqual(3) // 2 days + total

    // Change the first day places
    await user.clear(inputs[0])
    await user.type(inputs[0], '8')
    await flushPromises()

    expect(handleDayPlacesChange).toHaveBeenCalledWith(expect.any(Date), 8)
    expect(handleDayPlacesChange.mock.calls[0][0].toISOString().split('T')[0]).toBe('2021-02-10')
  })

  it('should call handlePlacesChange when total places are changed', async () => {
    const handleDayPlacesChange = jest.fn()
    const handlePlacesChange = jest.fn()

    const { user } = renderWithUserEvents(
      <DayPlacesTable
        event={eventWithPlacesPerDay}
        disabled={false}
        handleDayPlacesChange={handleDayPlacesChange}
        handlePlacesChange={handlePlacesChange}
        totalEnabled={true}
      />,
      undefined,
      {
        advanceTimers: jest.advanceTimersByTime,
      }
    )
    await flushPromises()

    const inputs = screen.getAllByRole('textbox')
    const totalInput = inputs[inputs.length - 1]

    await user.clear(totalInput)
    await user.type(totalInput, '15')
    await flushPromises()

    expect(handlePlacesChange).toHaveBeenCalledWith(15)
  })

  it('should disable day inputs when totalEnabled=true', () => {
    const handleDayPlacesChange = jest.fn()
    const handlePlacesChange = jest.fn()

    render(
      <DayPlacesTable
        event={eventWithPlacesPerDay}
        disabled={false}
        handleDayPlacesChange={handleDayPlacesChange}
        handlePlacesChange={handlePlacesChange}
        totalEnabled={true}
      />
    )

    const inputs = screen.getAllByRole('textbox')
    const [day1, day2, total] = inputs

    // Day inputs should be disabled, total should be enabled
    expect(day1).toBeDisabled()
    expect(day2).toBeDisabled()
    expect(total).toBeEnabled()
  })

  it('should disable total input when totalEnabled=false', () => {
    const handleDayPlacesChange = jest.fn()
    const handlePlacesChange = jest.fn()

    render(
      <DayPlacesTable
        event={eventWithPlacesPerDay}
        disabled={false}
        handleDayPlacesChange={handleDayPlacesChange}
        handlePlacesChange={handlePlacesChange}
        totalEnabled={false}
      />
    )

    const inputs = screen.getAllByRole('textbox')
    const [day1, day2, total] = inputs

    // Day inputs should be enabled, total should be disabled
    expect(day1).toBeEnabled()
    expect(day2).toBeEnabled()
    expect(total).toBeDisabled()
  })
})
