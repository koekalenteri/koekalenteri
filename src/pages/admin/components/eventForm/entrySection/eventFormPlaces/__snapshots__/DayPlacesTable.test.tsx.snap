// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayPlacesTable should render correctly with totalEnabled=false 1`] = `
<div>
  <table
    class="MuiTable-root css-1nkh8i9-MuiTable-root"
  >
    <thead
      class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head css-1xtozoh-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall css-cdeinc-MuiTableCell-root"
          scope="col"
        >
          date
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeSmall css-m4o53m-MuiTableCell-root"
          scope="col"
        >
          Paikat
        </th>
      </tr>
    </thead>
    <tbody
      class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root css-1xtozoh-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-jiabcu-MuiTableCell-root"
          scope="row"
        >
          eeeeee d.M.
        </th>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall css-wsxt5g-MuiTableCell-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
            inputmode="numeric"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1pzfmz2-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r3:"
                pattern="[0-9]{1,3}"
                style="text-align: right; padding: 4px;"
                type="text"
                value="5"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1nf2c5d-MuiNotchedOutlined-root"
                >
                  <span
                    aria-hidden="true"
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1xtozoh-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-jiabcu-MuiTableCell-root"
          scope="row"
        >
          eeeeee d.M.
        </th>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall css-wsxt5g-MuiTableCell-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
            inputmode="numeric"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1pzfmz2-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r4:"
                pattern="[0-9]{1,3}"
                style="text-align: right; padding: 4px;"
                type="text"
                value="5"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1nf2c5d-MuiNotchedOutlined-root"
                >
                  <span
                    aria-hidden="true"
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1xtozoh-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-jiabcu-MuiTableCell-root"
          scope="row"
        >
          Yhteensä
        </th>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall css-wsxt5g-MuiTableCell-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
            inputmode="numeric"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-sizeSmall css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputSizeSmall css-1pzfmz2-MuiInputBase-input-MuiOutlinedInput-input"
                disabled=""
                id="event.places"
                pattern="[0-9]{1,3}"
                style="text-align: right; padding: 4px;"
                type="text"
                value="10"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1nf2c5d-MuiNotchedOutlined-root"
                >
                  <span
                    aria-hidden="true"
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`DayPlacesTable should render correctly with totalEnabled=true 1`] = `
<div>
  <table
    class="MuiTable-root css-1nkh8i9-MuiTable-root"
  >
    <thead
      class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head css-1xtozoh-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall css-cdeinc-MuiTableCell-root"
          scope="col"
        >
          date
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeSmall css-m4o53m-MuiTableCell-root"
          scope="col"
        >
          Paikat
        </th>
      </tr>
    </thead>
    <tbody
      class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root css-1xtozoh-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-jiabcu-MuiTableCell-root"
          scope="row"
        >
          eeeeee d.M.
        </th>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall css-wsxt5g-MuiTableCell-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
            inputmode="numeric"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-sizeSmall css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputSizeSmall css-1pzfmz2-MuiInputBase-input-MuiOutlinedInput-input"
                disabled=""
                id=":r0:"
                pattern="[0-9]{1,3}"
                style="text-align: right; padding: 4px;"
                type="text"
                value="5"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1nf2c5d-MuiNotchedOutlined-root"
                >
                  <span
                    aria-hidden="true"
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1xtozoh-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-jiabcu-MuiTableCell-root"
          scope="row"
        >
          eeeeee d.M.
        </th>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall css-wsxt5g-MuiTableCell-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
            inputmode="numeric"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-sizeSmall css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputSizeSmall css-1pzfmz2-MuiInputBase-input-MuiOutlinedInput-input"
                disabled=""
                id=":r1:"
                pattern="[0-9]{1,3}"
                style="text-align: right; padding: 4px;"
                type="text"
                value="5"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1nf2c5d-MuiNotchedOutlined-root"
                >
                  <span
                    aria-hidden="true"
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </td>
      </tr>
      <tr
        class="MuiTableRow-root css-1xtozoh-MuiTableRow-root"
      >
        <th
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-jiabcu-MuiTableCell-root"
          scope="row"
        >
          Yhteensä
        </th>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall css-wsxt5g-MuiTableCell-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1xp5r68-MuiFormControl-root-MuiTextField-root"
            inputmode="numeric"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1pzfmz2-MuiInputBase-input-MuiOutlinedInput-input"
                id="event.places"
                pattern="[0-9]{1,3}"
                style="text-align: right; padding: 4px;"
                type="text"
                value="10"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1nf2c5d-MuiNotchedOutlined-root"
                >
                  <span
                    aria-hidden="true"
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
